pmd-test/src/main/java/net/sourceforge/pmd/test/RuleTst.java

```diff
@@ -57,10 +57,19 @@ protected void setUp() {
         // This method is intended to be overridden by subclasses.
     }
 
+    /**
+     * Return the rules that will be tested. Each rule must have a corresponding XML file containing a test collection.
+     * Test collections for all these rules are run separately.
+     */
     protected List<Rule> getRules() {
         return Collections.emptyList();
     }
 
+    /** Return extra rules that will be run while running the tests. */
```

Comment:
```suggestion
    /** Return extra rules that will be run before the rule of the current test. */
```

Reply:
They are not necessarily run before the tested rule. All rules are run together. In this specific case there is an ordering, but the suppress warnings rule is the only one that has a specified ordering wrt other rules.